test/upgrade/prober/wathola/sender/services.go

```diff
@@ -101,15 +102,15 @@ func (s *sender) SendContinually() {
 			if retry == 0 {
 				start = time.Now()
 			}
-			log.Warnf("Could not send step event %v, retrying (%d)",
-				s.eventsSent, retry)
-			log.Debug("Error: ", err)
+			log.Warnf("Could not send step event %v, retrying (%d): %v",
+				s.eventsSent, retry, err)
 			retry++
 		} else {
 			if retry != 0 {
 				s.unavailablePeriods = append(s.unavailablePeriods, event.UnavailablePeriod{
-					Step:   currentStep,
-					Period: time.Since(start),
+					Step:    currentStep,
+					Period:  time.Since(start),
+					LastErr: err.Error(),
```

Comment:
I'd say the `err` variable is always nil at this point. So that probably causes the problems? You could probably define another `var lastErr error` before the loop and in each iteration at the beginning do `lastErr = err` . And then use `LastErr: lastErr` instead of `LastErr: err.Error()`

Reply:
good catch, thanks!